app/helpers/distribution_helper.rb

```diff
@@ -20,12 +20,22 @@ def markdownify(text)
 
   def retrieve_image_size(url)
     conn = Faraday.new(url: url) do |f|
-      f.use FaradayMiddleware::FollowRedirects
+      f.use FaradayMiddleware::FollowRedirects, limit: 5
       f.request  :url_encoded
       f.adapter  Faraday.default_adapter
     end
     # Turn into integer just in case we have dead links (they will report 0B)
     conn.head.headers['content-length'].to_i
+  rescue Faraday::Error::ClientError => e
+    Rails.logger.error("Exception in distribution_helper#retrieve_image_size: #{e}")
+    0
+  end
+
+  def cached_image_size(name, url)
+    cache_key = ActiveSupport::Cache.expand_cache_key('image_size', name)
```

Comment:
[nitpick] why not adding the cache of the image in `retrieve_image_size_url`? I find a bit wierd that the view call `cached_image_size`. The view want just the image size, how this is done is logic that doesn't IMO belong to the view.

Reply:
Actually, I think I move more logic from the view to the helper, so all that is left in the view are these two lines:
```erb
              <h5 class="mb-2 text-muted"><%= number_to_human_size(image_size(type)) %></h5>
              <h6 class="mb-2 text-muted"><%= short_description(type) %></h6>

```